### PR TITLE
new: BAAI/BGE-M3 support with testing script

### PR DESCRIPTION
**Pull request by lucifertrj**
### All Submissions:

* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### New models submission:

* [x] Have you added an explanation of why it's important to include this model?
* [x] Have you added tests for the new model? Were canonical values for tests computed via the original model?
* [x] Have you added the code snippet for how canonical values were computed?
* [x] Have you successfully ran tests with your changes locally?

#### New Model: BAAI/bge-m3

MIT-Licensed:

| Model Name |  Dimension | Sequence Length | Introduction |
|:----:|:---:|:---:|:---:|
| [BAAI/bge-m3](https://huggingface.co/BAAI/bge-m3) | 1024 | 8192 | multilingual; unified fine-tuning (dense, sparse, and colbert) from bge-m3-unsupervised|

🔗 Colab Notebook: [Open In Colab](https://colab.research.google.com/drive/15nOo-KKArOc7m1uOEaiwz5N_78H8Er7T?usp=sharing)

#### Code Snippet for Canonical Values:

```py
docs = ["hello world", "flag embedding"]
embeddings = list(embedding_model.embed(docs))
embeddings = np.stack(embeddings, axis=0)

canonical = np.round(embeddings[0, :5], 4)
print(f"Canonical vector values: {canonical}")
```

Output:
```bash
Canonical vector values: [-0.0404  0.037  -0.029   0.0161 -0.0357]
```

Added these values in: ```tests/test_text_onnx_embeddings.py```

### CODE REVIEW

**Review by JiwaniZakir — COMMENTED** [review]
The `DenseModelDescription` in `onnx_embedding.py` advertises "8192 input tokens truncation" in the description, but there's no corresponding `additional_kwargs` (e.g., `{"max_length": 8192}`) or similar field to actually configure the tokenizer's max sequence length. Without this, the tokenizer will fall back to its default, likely 512 tokens, silently discarding the model's long-context capability and making the description misleading. It's worth checking how other long-context models in this registry handle that setting.

The test entry in `test_text_onnx_embeddings.py` only checks 5 embedding dimensions, consistent with the rest of the suite, but there's no assertion that the output shape matches the declared `dim=1024`. Given the model is 2.27 GB and has an unusual external data file (`model.onnx_data`), a shape check would help catch loading issues early — for instance, if the wrong ONNX graph is loaded or the external data file is missing at inference time.

**Comment by yerffejytnac** [comment]
Any news?

**Comment by X-Cotang** [comment]
Any update?

**Comment by ejohb** [comment]
This seems like a really important model to support natively, given it encodes all 3 vector types in a single pass.

**Comment by Dylancouzon** [comment]
Thanks to everyone who contributed here, and apologies for the delay. This one slipped through the cracks.

### What I changed

1. Merged `main`
2. Reworded the description to match the sibling entries.
3. Removed `onnx/sentencepiece.bpe.model` from `additional_files`. `fastembed` reads the root `tokenizer.json`, so only `onnx/model.onnx_data` is needed.
4. Recomputed the canonical values from the original PyTorch model, since the previous ones were generated with fastembed
```py
from transformers import AutoTokenizer, AutoModel
import torch, torch.nn.functional as F

tok = AutoTokenizer.from_pretrained("BAAI/bge-m3")
model = AutoModel.from_pretrained("BAAI/bge-m3").eval()

inp = tok(["hello world", "flag embedding"], padding=True, truncation=True,
          max_length=8192, return_tensors="pt")
with torch.no_grad():
    emb = F.normalize(model(**inp).last_hidden_state[:, 0], p=2, dim=1)
print(emb[0, :5])
# tensor([-0.0403941, 0.03703506, -0.02897445, 0.01611726, -0.03569157])
```

### Verification

`ruff`, `mypy`, `pyright`, and the test suite all pass. `fastembed` matches the PyTorch reference to `1.2e-7` on both Apple Silicon and `linux/amd64`, and the two platforms agree to `3.9e-8`.

Two things for you @joein:

Which class should own this: `OnnxTextEmbedding` or `BuiltinSentenceEmbedding`? 

At 2.27 GB, the model is skipped by local runs and lightweight CI. The only way to run it in CI is via a manual `workflow_dispatch`
